src/jtag3.c

```diff
@@ -2310,8 +2344,10 @@ static int jtag3_write_byte(const PROGRAMMER *pgm, const AVRPART *p, const AVRME
       unsupp = 1;
   }
 
-  if (unsupp)
+  if (unsupp) {
+    pmsg_error("debugWire interface does not support writing to memory %s\n", mem->desc);
```

Comment:
Would be better to complain only with error if the user wants to *change* the data. We should return OK if data is unchanged to allow a user to read a read-only memory and write it back without error. This in preparation of a future functionality of backup the entire device and restore it. Otherwise we get write errors on read-only memories, So, better to read contents here and if the same as to be written return 0.